### PR TITLE
TT-11537 added new MDCB config option healthcheck.cache_renewal_period

### PR DESCRIPTION
**Pull request by sredxny**
## Description
Enable new configurations `healthcheck.cache_renewal_period` via helm chart. It should set corresponding MDCB environment variable.

Modify default MDCB probes to use new /liveness and /readiness endpoint exposed (ref: [tyk-charts/components/tyk-mdcb/values.yaml at 15141e28425b5ddd9d393e7892f864dbe1cca7be · TykTechnologies/tyk-charts](https://github.com/TykTechnologies/tyk-charts/blob/15141e28425b5ddd9d393e7892f864dbe1cca7be/components/tyk-mdcb/values.yaml#L214) )

Add optional Ingress resource for the new health endpoints. It allows users to access the liveness and readiness check from outside the cluster.

Current [ingress](https://github.com/TykTechnologies/tyk-charts/blob/15141e28425b5ddd9d393e7892f864dbe1cca7be/components/tyk-mdcb/values.yaml#L309) configuration expose port 9091 (the listenPort), which is used for rpc connection. We should create a new ingress configuration and resource that expose port 8181 (the httpPort) instead, for excessing HTTP services.

## Related Issue
TT-11537

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: components/tyk-mdcb/values.yaml [hunk 1]
```diff
@@ -233,7 +237,7 @@ mdcb:
     # readiness includes details about readiness probe used in MDCB Deployment.
     readiness:
       # path represents the http path to be used in readiness probe in MDBC deployment.
-      path: "/health"
+      path: "/readiness"
```
Comment (sredxny) [inline components/tyk-mdcb/values.yaml:240]: am wondering if in https://github.com/TykTechnologies/tyk-charts/pull/306/files#diff-b8f26e71ad3157bc8ed4f570bcc62cabdd9a80e016ad01e39db77137cb0c67a5R50
we should do something like:

`path: {{- if  semverCompare ">=2.6.0" $version }}/liveness{{ else }}/health{{ end }}`

Comment (olamilekan000) [reply components/tyk-mdcb/values.yaml:240]: it's nice if we do it for backward compatibility

File: components/tyk-mdcb/values.yaml [hunk 2]
```diff
@@ -303,7 +307,17 @@ mdcb:
     externalTrafficPolicy: Local
     # annotations corresponds to the annotations which will be added into Tyk Dashboard Service.
     annotations: {}
-
+  
+  http_service:
```
Comment (komalsukhani) [inline components/tyk-mdcb/values.yaml:311]: You also need to create service.

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: components/tyk-mdcb/values.yaml (hunk 2 above)
Comment (komalsukhani) [reply components/tyk-mdcb/values.yaml:311]: @sredxny You can skip creation of service as http port is already exposed in a service.
Please remove `http_service` section from values file and update PR accordingly.

Comment (sredxny) [reply components/tyk-mdcb/values.yaml]: thank you

**Comment by sredxny** [comment]
@komalsukhani I made the requested changes, could you take another look to the pr?

File: components/tyk-mdcb/templates/http-ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{{- if .Values.mdcb.http_ingress.enabled -}}
+{{- $fullName := printf (include "tyk-mdcb.fullname" .) -}}
+{{- $svcPort := .Values.mdcb.httpPort -}}
```
Comment (komalsukhani) [inline components/tyk-mdcb/templates/http-ingress.yaml:3]: Can you use `{{ include "mdcb.healthCheckPort" . }}` here?

File: tyk-control-plane/values.yaml [hunk 4]
```diff
@@ -1224,7 +1224,10 @@ tyk-mdcb:
     # Users should use httpPort instead of probes.healthCheckPort for newer versions.
     # This value is used to set TYK_MDCB_HTTPPORT when MDCB is >= 2.6.0.
     httpPort: 8181
-
+    # settings for the health checker
+    healthcheck:
```
Comment (komalsukhani) [inline tyk-control-plane/values.yaml:1228]: Can you update liveliness and readiness probes in this file too?
Also settings for http_ingress is missing here.

**Review by komalsukhani — CHANGES_REQUESTED** [review]

**Review by komalsukhani — APPROVED** [review]

**Comment by komalsukhani** [comment]
@sredxny Thank you for the PR. I have approved it. 
After your QA test the PR we can merge it.

**Comment by bojantyk** [comment]
hey @sredxny  @komalsukhani  can we update mdcb tag version in value.yaml file to 2.7.0 
 https://github.com/TykTechnologies/tyk-charts/pull/306/files#diff-1f1f1607e0420b817d40d38eba3a4a3d09a64e273cdffedc8d1136f0afdeff63L253

**Pull request merged**